In [1]:
#dependencies
import pandas as pd
import numpy as np

In [2]:
#load csv and prepare dataframe.
file = "Resources/purchase_data.csv"
purchase_data_df = pd.read_csv(file, delimiter = ',')
purchase_data_df.head(10)

Purchase ID             SN  Age                 Gender  Item ID  \
0            0        Lisim78   20                   Male      108   
1            1    Lisovynya38   40                   Male      143   
2            2     Ithergue48   24                   Male       92   
3            3  Chamassasya86   24                   Male      100   
4            4      Iskosia90   23                   Male      131   
5            5        Yalae81   22                   Male       81   
6            6      Itheria73   36                   Male      169   
7            7    Iskjaskst81   20                   Male      162   
8            8      Undjask33   22                   Male       21   
9            9    Chanosian48   35  Other / Non-Disclosed      136   

                                   Item Name  Price  
0  Extraction, Quickblade Of Trembling Hands   3.53  
1                          Frenzied Scimitar   1.56  
2                               Final Critic   4.88  
3                                Blindscythe   3.27  
4                                       Fury   1.44  
5                                  Dreamkiss   3.61  
6     Interrogator, Blood Blade of the Queen   2.18  
7                              Abyssal Shard   2.67  
8                                  Souleater   1.10  
9               Ghastly Adamantite Protector   3.58

In [3]:
# Player Count
total_players_df = purchase_data_df.groupby(purchase_data_df['SN'])
total_player_len = len(total_players_df)

# print statement: total players
print(f' The total number of players is: {total_player_len}')

#output total players as df
tpl_df = pd.DataFrame({'Total Players': total_player_len}, 
                      index=[0])
tpl_df

 The total number of players is: 576


Total Players
0            576

In [4]:
# Purchasing Analysis (Total)
# Number of Unique Items
unique_items_df = purchase_data_df.groupby(purchase_data_df['Item ID'])
unique_item_count = len(unique_items_df)

# Average Purchase Price
avg_price = round(purchase_data_df['Price'].mean(), 2)

# Total Number of Purchases
total_purchases = len(purchase_data_df['Purchase ID'])

# Total Revenue
total_revenue = purchase_data_df['Price'].sum()
total_revenue =  float(total_revenue)

# presentation dataframe
purchase_analysis_df = pd.DataFrame({'Number of Unique Items': unique_item_count,
                                    'Average Price': avg_price,
                                    'Number of Purchases': total_purchases,
                                    'Total Revenue': total_revenue},
                                   index=[0])

#format purchase analysis dataframe
purchase_analysis_df['Average Price'] = purchase_analysis_df['Average Price'].map('${:,.2f}'.format)
purchase_analysis_df['Total Revenue'] = purchase_analysis_df['Total Revenue'].map('${:,.2f}'.format)


purchase_analysis_df

Number of Unique Items Average Price  Number of Purchases Total Revenue
0                     183         $3.05                  780     $2,379.77

In [5]:
# Gender Demographics

unique_users = purchase_data_df.copy()
unique_users.drop_duplicates(subset ="SN", keep = 'first', inplace = True) 
user_count = len(unique_users)

gender_df = unique_users.groupby(unique_users['Gender'])
gender_count = gender_df.count()
gender_count = gender_count.reset_index(drop=False)

# Percentage and Count of Male Players

male_count = gender_count.iloc[1,1]
male_percent = round(male_count / user_count * 100 , 2)


# Percentage and Count of Female Players

female_count = gender_count.iloc[0,1]
female_percent = round(female_count / user_count * 100 , 2)


# Percentage and Count of Other / Non-Disclosed
ndgender_count = gender_count.iloc[2,1]
ndgender_percent = round(ndgender_count / user_count * 100 , 2)

# Gender Demographics
gender_demographic_df = pd.DataFrame({'Total Count':[male_count, female_count, ndgender_count],
                                        'Percent of Players':[male_percent, female_percent, ndgender_percent]},
                                        index = ['Male','Female','Other/Non-Disclosed'])
gender_demographic_df

Total Count  Percent of Players
Male                         484               84.03
Female                        81               14.06
Other/Non-Disclosed           11                1.91

In [6]:
# Purchasing Analysis (Gender)
# The below each broken by gender

# Purchase Count
gender_purchase = purchase_data_df.groupby(purchase_data_df['Gender'])
gender_purchase_count = gender_purchase.count()

male_purchase_count = gender_purchase_count.iloc[1,1]
female_purchase_count = gender_purchase_count.iloc[0,1]
ndgender_purchase_count = gender_purchase_count.iloc[2,1]


# Average Purchase Price
gender_purchase_new = gender_purchase['Gender','Price']
gender_purchase_totals = gender_purchase_new.sum()

male_purchase_total = gender_purchase_totals.iloc[1,0]
avg_male_purchase = round(gender_purchase_totals.iloc[1,0] / male_purchase_count, 2)

female_purchase_total = gender_purchase_totals.iloc[0,0]
avg_female_purchase = round(gender_purchase_totals.iloc[0,0] / female_purchase_count, 2)

ndgender_purchase_total = gender_purchase_totals.iloc[2,0]
avg_ndgender_purchase = round(gender_purchase_totals.iloc[2,0] / ndgender_purchase_count, 2)


# Total Purchase Value
# See above under (gendername)_purchase_total


# Average Purchase Total per Person by Gender
avg_PTPP_male = round(male_purchase_total / male_count,2)
avg_PTPP_female = round(female_purchase_total / female_count,2)
avg_PTPP_ndgender = round(ndgender_purchase_total / ndgender_count,2)


# Gender Purchases
gender_purchases_df = pd.DataFrame({'Purchase Count':[female_purchase_count, male_purchase_count, ndgender_purchase_count],
                                        'Average Purchase Price':[avg_female_purchase, avg_male_purchase, avg_ndgender_purchase],
                                    'Total Purchase Value':[female_purchase_total, male_purchase_total, ndgender_purchase_total],
                                       'Average Purchase Total per Person':[avg_PTPP_female, avg_PTPP_male, avg_PTPP_ndgender]},
                                        index = ['Female','Male','Other/Non-Disclosed'])
#format
gender_purchases_df['Average Purchase Price'] = gender_purchases_df['Average Purchase Price'].map('${:,.2f}'.format)
gender_purchases_df['Total Purchase Value'] = gender_purchases_df['Total Purchase Value'].map('${:,.2f}'.format)
gender_purchases_df['Average Purchase Total per Person'] = gender_purchases_df['Average Purchase Total per Person'].map('${:,.2f}'.format)

gender_purchases_df

Purchase Count Average Purchase Price  \
Female                          113                  $3.20   
Male                            652                  $3.02   
Other/Non-Disclosed              15                  $3.35   

                    Total Purchase Value Average Purchase Total per Person  
Female                           $361.94                             $4.47  
Male                           $1,967.64                             $4.07  
Other/Non-Disclosed               $50.19                             $4.56

In [7]:
# Age Demographics
# The below each broken into bins of 4 years (i.e. <10, 10-14, 15-19, etc.)

# Bin the purchase_data data frame by age
bins= [0,9,14,19,24,29,34,39,44,49,200]
age_cats = ['<10', '10-14', '15-19', '20-24','25-29','30-34','35-39','40-44','45-49','50+']

age_group_df = purchase_data_df.copy()
age_group_df["Age Groups"] = pd.cut(age_group_df["Age"], bins, labels=age_cats)

age_groupby = age_group_df.groupby(age_group_df['Age Groups'])
age_groupby_count = age_groupby['Price'].count()
age_groupby_count

age_total = age_group_df.iloc[:,0].count()
age_percent = round(age_groupby_count / age_total * 100, 2)


# #summary
# summary_age = pd.DataFrame({'Total Players':age_groupby_count,'Percentage of Players':age_percent},index=[age_cats])
summary_age = age_groupby_count.to_frame()
summary_age['Percentage of Players'] = age_percent
summary_age = summary_age.rename(columns={"Price": "Total Count"})
summary_age

Total Count  Percentage of Players
Age Groups                                    
<10                  23                   2.95
10-14                28                   3.59
15-19               136                  17.44
20-24               365                  46.79
25-29               101                  12.95
30-34                73                   9.36
35-39                41                   5.26
40-44                12                   1.54
45-49                 1                   0.13
50+                   0                   0.00

In [13]:
#Purchase Analysis (AGE)
purchase_analysis = summary_age.copy()
purchase_analysis = purchase_analysis.drop(columns='Percentage of Players')

# Purchase Count
purchase_analysis = purchase_analysis.rename(columns={"Total Count": "Purchase Count"})


# Average Purchase Price



# Total Purchase Value
age_total_purchase = age_groupby['Price'].sum()


purchase_analysis['Total Purchase Value'] = age_total_purchase
# age_people_per_group = age_groupby['Price'].count()


# Average Purchase Total per Person by Age Group
total_per_age = age_total_purchase / age_groupby_count
purchase_analysis['Avg Total Purchase per Person'] = round(total_per_age,2)
purchase_analysis = purchase_analysis.dropna()
purchase_analysis['Avg Total Purchase per Person'] = purchase_analysis['Avg Total Purchase per Person'].map('${:,.2f}'.format)
purchase_analysis['Total Purchase Value'] = purchase_analysis['Total Purchase Value'].map('${:,.2f}'.format)

purchase_analysis

Purchase Count Total Purchase Value Avg Total Purchase per Person
Age Groups                                                                   
<10                     23               $77.13                         $3.35
10-14                   28               $82.78                         $2.96
15-19                  136              $412.89                         $3.04
20-24                  365            $1,114.06                         $3.05
25-29                  101              $293.00                         $2.90
30-34                   73              $214.00                         $2.93
35-39                   41              $147.67                         $3.60
40-44                   12               $36.54                         $3.04
45-49                    1                $1.70                         $1.70

In [14]:
# Top Spenders
# Identify the the top 5 spenders in the game by total purchase value, then list (in a table):
# SN

df_top = purchase_data_df.groupby(['SN'])
df_top_count = df_top.count().sort_values('Purchase ID', ascending=False)
df_top_count.head(5)

# Total Purchase Value
top_tpv =  round(df_top['Price'].sum(),2)

# Average Purchase Price
top_count = df_top['Price'].count()
top_apv = round(top_tpv / top_count,2)


#summary DF
top_summary_df = pd.DataFrame(df_top_count['Purchase ID'])
top_summary_df = top_summary_df.rename(columns={"Purchase ID": "Purchase Count"})
top_summary_df['Average Purchase Price'] = top_apv
top_summary_df['Total Purchase Value'] =  top_tpv

#format df
top_summary_df['Average Purchase Price'] = top_summary_df['Average Purchase Price'].map('${:,.2f}'.format)
top_summary_df['Total Purchase Value'] = top_summary_df['Total Purchase Value'].map('${:,.2f}'.format)





top_summary_df.head()

Purchase Count Average Purchase Price Total Purchase Value
SN                                                                     
Lisosia93                 5                  $3.79               $18.96
Iral74                    4                  $3.40               $13.62
Idastidru52               4                  $3.86               $15.45
Asur53                    3                  $2.48                $7.44
Inguron55                 3                  $3.70               $11.11

In [15]:
# Most Popular Items
# Identify the 5 most popular items by purchase count, then list (in a table):


df_top_item = purchase_data_df.groupby(['Item Name','Item ID','Price'])
df_top_item_count = df_top_item.count().sort_values('Item ID', ascending=False)
# df_top_item_count.head(5)
# Item ID
item_id_df = df_top_item['Item ID']
# Item Name


# Purchase Count


# Item Price


# Total Purchase Value
df_top_item_count = df_top_item_count.drop(columns=['SN','Age','Gender'])
df_top_item_count = df_top_item_count.rename(columns={"Purchase ID": "Purchase Count"})
df_top_item_count['Total Purchase Value']= round(df_top_item['Price'].sum(),2)
df_top_item_count = df_top_item_count.reset_index(drop=False)
df_top_item_count = df_top_item_count.set_index('Item ID')


df_top_item_count2 = df_top_item_count.copy()


df_top_item_count['Price'] = df_top_item_count['Price'].map('${:,.2f}'.format)
df_top_item_count['Total Purchase Value'] = df_top_item_count['Total Purchase Value'].map('${:,.2f}'.format)


item_summary_df = df_top_item_count.sort_values('Purchase Count',ascending=False)
item_summary_df.head(5)

Item Name  Price  Purchase Count  \
Item ID                                                                        
178      Oathbreaker, Last Hope of the Breaking Storm  $4.23              12   
82                                            Nirvana  $4.90               9   
108         Extraction, Quickblade Of Trembling Hands  $3.53               9   
145                              Fiery Glass Crusader  $4.58               9   
92                                       Final Critic  $4.88               8   

        Total Purchase Value  
Item ID                       
178                   $50.76  
82                    $44.10  
108                   $31.77  
145                   $41.22  
92                    $39.04

In [16]:
# Most Profitable Items
# df_top_item_count2['Price'] = df_top_item_count2['Price'].map('${:,.2f}'.format)
# df_top_item_count2['Total Purchase Value'] = df_top_item_count2['Total Purchase Value'].map('${:,.2f}'.format)

item_summary_df_2 = df_top_item_count2.sort_values('Purchase Count',ascending=False)
item_summary_df_2['Price'] = item_summary_df_2['Price'].map('${:,.2f}'.format)
item_summary_df_2['Total Purchase Value'] = item_summary_df_2['Total Purchase Value'].map('${:,.2f}'.format)

item_summary_df_2.head(5)

Item Name  Price  Purchase Count  \
Item ID                                                                        
178      Oathbreaker, Last Hope of the Breaking Storm  $4.23              12   
82                                            Nirvana  $4.90               9   
108         Extraction, Quickblade Of Trembling Hands  $3.53               9   
145                              Fiery Glass Crusader  $4.58               9   
92                                       Final Critic  $4.88               8   

        Total Purchase Value  
Item ID                       
178                   $50.76  
82                    $44.10  
108                   $31.77  
145                   $41.22  
92                    $39.04

# Trends:

    
* The most notable tend would be that the age 20-24 group spent ($1,114.06) over twice as much as any other age group.</li>
* The above tend also correlates with the fact that almost half of the players in the game are from this age group. </li>
* Finally, over 80 percent of the players in the game are male compared to females or nondisclosed gender individuals.(</li>

